nav2_bringup/launch/developer_launch.py

```diff
@@ -0,0 +1,225 @@
+    map_yaml_file = launch.substitutions.LaunchConfiguration('map')
+    params_file = launch.substitutions.LaunchConfiguration('params')
+    rviz_config_file = launch.substitutions.LaunchConfiguration('rviz_config')
+    simulator = launch.substitutions.LaunchConfiguration('simulator')
+    use_sim_time = text = launch.substitutions.LaunchConfiguration('use_sim_time')
+    use_simulation = launch.substitutions.LaunchConfiguration('use_simulation')
+    world = launch.substitutions.LaunchConfiguration('world')
+
+    # Create our own temporary YAML files that include the following parameter substitutions
+    param_substitutions = {
+        'autostart': autostart,
+        'bt_xml_filename': bt_xml_file,
+        'use_sim_time': use_sim_time,
+        'yaml_filename': map_yaml_file
+    }
+    configured_params = RewrittenYaml(
+        source_file=params_file, rewrites=param_substitutions, convert_types=True)
+
+    # Declare the launch arguments
+    declare_autostart_cmd = launch.actions.DeclareLaunchArgument(
+        'autostart',
+        default_value='false',
+        description='Automatically startup the nav2 stack')
+
+    declare_bt_xml_cmd = launch.actions.DeclareLaunchArgument(
+        'bt',
+        default_value=os.path.join(
+            get_package_prefix('nav2_bt_navigator'),
+            'behavior_trees/navigate_w_replanning.xml'),
+        description='Full path to the Behavior Tree XML file to use for the BT navigator')
+
+    declare_map_yaml_cmd = launch.actions.DeclareLaunchArgument(
+        'map',
+        default_value='turtlebot3_world.yaml',
+        description='Full path to map file to load')
+
+    declare_params_file_cmd = launch.actions.DeclareLaunchArgument(
+        'params',
+        default_value=[launch.substitutions.ThisLaunchFileDir(), '/nav2_params.yaml'],
+        description='Full path to the ROS2 parameters file to use for all launched nodes')
+
+    declare_rviz_config_file_cmd = launch.actions.DeclareLaunchArgument(
+        'rviz_config',
+        default_value='nav2_default_view.rviz',
+        description='Full path to the RVIZ config file to use')
+
+    declare_simulator_cmd = launch.actions.DeclareLaunchArgument(
+        'simulator',
+        default_value='gazebo',
+        description='The simulator to use (gazebo or gzserver)')
+
+    declare_use_sim_time_cmd = launch.actions.DeclareLaunchArgument(
+        'use_sim_time',
+        default_value='true',
+        description='Use simulation (Gazebo) clock if true')
+
+    declare_use_simulation_cmd = launch.actions.DeclareLaunchArgument(
+        'use_simulation', condition=IfCondition('True'),
+        default_value='True',
+        description='Whether to run in simulation')
+
+    declare_world_cmd = launch.actions.DeclareLaunchArgument(
+        'world',
+        default_value=os.path.join(get_package_share_directory('turtlebot3_gazebo'),
+                                   'worlds/turtlebot3_worlds/burger.model'),
```

Comment:
We should default to the waffle instead of the burger, since that is the robot we actually have. Also, it is the one the params are configured for.

Reply:
Ok, fixed.